src/main/java/uk/gov/hmcts/divorce/citizen/event/CitizenRespondToRequestForInformation.java

```diff
@@ -0,0 +1,219 @@
+
+    public static final String CITIZEN_RESPOND_TO_REQUEST_FOR_INFORMATION = "citizen-respond-request-for-information";
+    public static final String UNABLE_TO_DETERMINE_CITIZEN_ERROR = "Unable to determine citizen for Case Id: ";
+    public static final String CITIZEN_NOT_VALID_FOR_PARTY_START_ERROR = "Unable to apply response. Applicant ";
+    public static final String CITIZEN_NOT_VALID_FOR_PARTY_MID_ERROR = "not valid for Party (";
+    public static final String CITIZEN_NOT_VALID_FOR_PARTY_END_ERROR = ") on latest RFI for Case Id: ";
+    public static final String REQUEST_FOR_INFORMATION_RESPONSE_NOTIFICATION_FAILED_ERROR
+        = "Request for Information Response Notification for Case Id {} failed with message: {}";
+    public static final String REQUEST_FOR_INFORMATION_RESPONSE_PARTNER_NOTIFICATION_FAILED_ERROR
+        = "Request for Information Response Partner Notification for Case Id {} failed with message: {}";
+
+    private final CcdAccessService ccdAccessService;
+    private final HttpServletRequest request;
+    private final NotificationDispatcher notificationDispatcher;
+    private final CitizenRequestForInformationResponseNotification citizenRequestForInformationResponseNotification;
+    private final CitizenRequestForInformationResponsePartnerNotification citizenRequestForInformationResponsePartnerNotification;
+
+    @Override
+    public void configure(final ConfigBuilder<CaseData, State, UserRole> configBuilder) {
+        new PageBuilder(configBuilder
+            .event(CITIZEN_RESPOND_TO_REQUEST_FOR_INFORMATION)
+            .forState(InformationRequested)
+            .name("Submit response for rfi")
+            .description("Submit response for RFI")
+            .showSummary()
+            .showEventNotes()
+            .aboutToSubmitCallback(this::aboutToSubmit)
+            .submittedCallback(this::submitted)
+            .grant(CREATE_READ_UPDATE_DELETE, CREATOR, APPLICANT_2)
+            .grantHistoryOnly(
+                CASE_WORKER,
+                SUPER_USER,
+                LEGAL_ADVISOR,
+                JUDGE))
+            .page("respondRequestForInfo")
+            .pageLabel("Submit response for RFI");
+    }
+
+    public AboutToStartOrSubmitResponse<CaseData, State> aboutToSubmit(final CaseDetails<CaseData, State> details,
+                                                                       final CaseDetails<CaseData, State> beforeDetails) {
+
+        log.info("{} about to submit callback invoked for Case Id: {}", CITIZEN_RESPOND_TO_REQUEST_FOR_INFORMATION, details.getId());
+
+        final RequestForInformationList requestForInformationList = details.getData().getRequestForInformationList();
+        final RequestForInformationResponse response = new RequestForInformationResponse();
+        RequestForInformationResponseParties responseParty;
+        if (isApplicant1(details.getId())) {
+            responseParty = APPLICANT1;
+        } else if (isApplicant2(details.getId())) {
```

Comment:
Nitpick: When it gets to the `else if` here I believe we will have sent the request to get the users roles twice (via `ccdAccessService`). Probably won't make a massive difference, but refactoring to get the roles once or turning it into if/else could improve performance for the citizen by cutting down the number of requests we send to other APIs.

I'm not sure it will ever get to the error message because the user has to be CREATOR or APPLICANT_2 to use the event?

Reply:
Yeah ok - I may have been overly defensive here.  So we can safely assume that the request will always be from a user with the CREATOR or APPLICANT2 roles, therefore if not Applicant1, it will be Applicant2.  I'll refactor and remove the error.